src/lib/db/clients/sqlite.js

```diff
@@ -124,6 +130,10 @@ export async function applyChanges(conn, changes) {
     await driverExecuteQuery(cli, { query: 'BEGIN'})
 
     try {
+      if (changes.inserts) {
```

Comment:
One thing that I think would be very helpful (this has the ability to wreck someone's database if there's a bug) would be to have these functions covered by the integration test suite.

Covering:
- inserting good data
- Trying to insert a bad row (transaction shouldn't commit)
- delete, update, and inserts mixed together, should do the right thing
- Bad insert or update mixed with a good version of another (transaction should revert it)